File: src/main/java/com/ttknp/logic_layer/entities/Author.java
package com.ttknp.logic_layer.entities;

import jakarta.persistence.*;
import lombok.Data;
import java.util.List;

// @AllArgsConstructor
// @NoArgsConstructor
@Entity
@Table(name = "authors")
@Data
public class Author {
    @Id
    private String aid;
    private String fullname;
    private Short age;
    private Boolean alive;
    // Mapping to the other table
    @OneToMany(cascade = CascadeType.ALL) // ,targetEntity = Address.class ,mappedBy = "ob"
    @JoinColumn(name = "aid")
    private List<EditHistory> editHistories; // ** it's not good for toString()

    public Author() {}

    @Override
    public String toString() {
        return "Author{" +
                "aid='" + aid + '\'' +
                ", fullname='" + fullname + '\'' +
                ", age=" + age +
                ", alive=" + alive +
                '}';
    }
}

File: src/app/entity/author.ts
import {EditHistories} from "./edit-history";

export class Author {
  public aid: string
  public fullname: string
  public age: number
  public alive: boolean
  public image!: string
  public editHistories: EditHistories[]

  constructor(aid: string, fullname: string, age: number, alive: boolean, editHistories: EditHistories[]) {
    this.aid = aid;
    this.fullname = fullname;
    this.age = age;
    this.alive = alive;
    this.editHistories = editHistories;
  }
}
